numpy/core/src/multiarray/multiarray_tests.c.src

```diff
@@ -1036,7 +1038,7 @@ array_solve_diophantine(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject
         return NULL;
     }
 
-    if (PyTuple_GET_SIZE(A) > sizeof(terms) / sizeof(diophantine_term_t)) {
+    if (PyTuple_GET_SIZE(A) > (Py_ssize_t)ARRAY_SIZE(terms)) {
```

Comment:
Casting the other way around would be safer, but unlikely to be worthwhile.

As an aside - I think we almost always use `np_intp` instead of `Py_ssize_t`, even if the latter is more correct. Perhaps we should do the same here.

Below you also seem to use `np_intp`

Reply:
I just copied the return type of `PyTuple_GET_SIZE`. I suppose we could cast it to `size_t`, which is what was happening before, OTOH, I already know the size of `terms` and that the cast is safe. I'd probably just cast it to `int` myself.